tensorflow/c/kernels/summary_op.cc

```diff
@@ -0,0 +1,177 @@
+
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <sstream>
+#include <string>
+
+#include "tensorflow/c/kernels/tensor_shape_utils.h"
+#include "tensorflow/c/kernels.h"
+#include "tensorflow/c/tf_tensor.h"
+#include "tensorflow/c/tf_status.h"
+#include "tensorflow/core/platform/protobuf.h"
+#include "tensorflow/core/platform/logging.h"
+#include "tensorflow/core/platform/macros.h"
+#include "tensorflow/core/platform/tstring.h"
+#include "tensorflow/core/platform/strcat.h"
+#include "tensorflow/core/framework/selective_registration.h"
+#include "tensorflow/core/framework/summary.pb.h"
+#include "tensorflow/core/framework/types.h"
+
+namespace {
+
+// Struct that stores the status and TF_Tensor inputs to the opkernel. 
+// Used to delete tensor and status in its destructor upon kernel return. 
+struct Params { 
+  TF_Tensor* tags; 
+  TF_Tensor* values; 
+  TF_Status* status; 
+  Params(TF_OpKernelContext* ctx) : tags(nullptr), 
```

Comment:
Mark `explicit`. We require it for single-argument constructors.
`Params(TF_OpKernelContext* ctx) : tags(nullptr), ` --> `explicit Params(TF_OpKernelContext* ctx) : tags(nullptr), `